types/tmi.js/index.d.ts

```diff
@@ -0,0 +1,223 @@
+// Type definitions for tmi.js 1.2.1
+// Project: https://github.com/tmijs/tmi.js
+// Definitions by: DyingAlbatross <https://github.com/dyingalbatross>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'tmi.js' {
+	var tmi: Tmi;
+
+	export = tmi;
+}
+
+interface UserState {
+  badges: any;
+  color: string;
+  'display-name': string;
+  emotes: any;
+  mod: boolean;
+  'room-id': string;
+  subscriber: boolean;
+  turbo: boolean;
+  'user-id': string;
+  'user-type': string;
+  'emotes-raw': string;
+  'badges-raw': string;
+  'username': string;
+  'message-type': string;
+}
+
+interface ChannelState {
+  'broadcaster-lang': string | null;
+  r9k: boolean;
+  slow: boolean;
+  'subs-only': boolean;
+  channel: string;
+}
+
+interface TmiOptions {
+  options?: { 
+    clientId?: string; 
+    debug?: boolean; 
+  };
+  connection?: {
+    server?: string;
+    port?: number;
+    reconnect?: boolean;
+    maxReconnectAttempts?: number;
+    maxReconnectInterval?: number;
+    reconnectDelay?: number;
+    reconnectInterval?: number;
+    secure?: boolean;
+    timeout?: number;
+  };
+  identity?: {
+    username: string;
+    password: string;
+  };
+  channels?: string[];
+  logger?: any;
+}
+
+interface Tmi {
+  client: ClientStatic;
+}
+
+interface ClientStatic {
+  new (options?: TmiOptions): Client;
+}
+
+interface ApiOptions {
+  url: string;
+  headers?: any;
+  method?: string;
+}
+
+interface Client {
```

Comment:
`client` is a constructor. You should make this into a class.

**Note**: lowercase c in `client`.